engines/bastion_katello/app/assets/javascripts/bastion_katello/tasks/task.factory.js

```diff
@@ -1,264 +1,112 @@
- /**
+/**
  * @ngdoc service
  * @name  Bastion.tasks.factory:Task
  *
  * @requires BastionResource
- * @requires $timeout
- * @requires $log
- * @requires CurrentOrganization
  *
  * @description
- *   Provides a BastionResource for task(s).
- *
- *   Also provides a polling bulk search for tasks, define in
- *   foreman-tasks engine API. This allows polling for multiple task
- *   statuses with a single request (e.g. currently running tasks of a
- *   user, recent tasks of a user, tasks for given resource, detail of
- *   one task etc). See `resource.registerSearch` and
- *   `resource.unregisterSearch` for details.
+ *   Provides a BastionResource for task or list of tasks.
  */
-
-angular.module('Bastion.tasks').factory('Task',
-    ['BastionResource', '$timeout', '$log', '$q', 'CurrentOrganization',
-    function (BastionResource, $timeout, $log, $q, CurrentOrganization) {
-        var bulkSearchRunning = false, searchIdGenerator = 0,
-            searchParamsById = {}, callbackById = {}, pollCount = 0, maxPollInterval = 10000;
-
-        var resource = BastionResource('/katello/api/v2/tasks/:id/:action',
-            {id: '@id', 'organization_id': CurrentOrganization}, {});
-
-        var foremanTasksResource = BastionResource('/foreman_tasks/api/tasks/:id/:action',
-            {},
+angular.module('Bastion.tasks').factory('Task', [
+    '$timeout',
+    'BastionResource',
+    'CurrentOrganization',
+    function($timeout, BastionResource, CurrentOrganization) {
+        var pendingTimeout;
+        var productId;
+
+        var resource = BastionResource(
+            '/foreman_tasks/api/tasks/',
+            {
+                'organization_id': CurrentOrganization,
+                isArray: true
+            },
             {
-                bulkSearch: {method: 'POST', isArray: true, params: { action: 'bulk_search'}}
+                update: { method: 'PUT' },
+                sync: { method: 'POST', params: { action: 'sync' } },
+                updateSyncPlan: {
+                    method: 'POST',
+                    params: { action: 'sync_plan' }
+                },
+                autocomplete: {
+                    method: 'GET',
+                    isArray: true,
+                    params: { id: 'auto_complete_search' }
+                }
             }
         );
 
-        function bulkSearchParams() {
-            var searches = [];
-            _.each(searchParamsById, function (searchParams, id) {
-                searchParams['search_id'] = id;
-                searches.push(searchParams);
-            });
-            return { searches: searches };
-        }
+        resource.cb = false;
```

Comment:
What does `cb` stand for and can we use a full word (or words) instead of an abbreviation?

Reply:
This is the callback from nutupane that's executed after the queryPaged is completed. I'm storing it here because we don't yet have the `product_id` which I think we can get rid of anyway.

Reply:
Ah okay, do you mind calling it `callback`?